src/Microsoft.Extensions.Logging.Analyzers/LogFormatAnalyzer.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Extensions.Logging.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class LogFormatAnalyzer : DiagnosticAnalyzer
+    {
+        public LogFormatAnalyzer()
+        {
+            SupportedDiagnostics = ImmutableArray.Create(new[]
+            {
+                Descriptors.MEL1NumericsInFormatString,
+                Descriptors.MEL2ConcatenationInFormatString,
+                Descriptors.MEL3FormatParameterCountMismatch
+            });
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(analysisContext =>
```

Comment:
Check if logging is even referenced. If not, skip the diagnostics altogether.

Reply:
Trying to find extensions type few lines lower does exactly this, I don't know if it can be done without having `Compilation`

Reply:
Yeah, you need compilation. The lines you added are fine.